Zend/zend_compile.h

```diff
@@ -456,8 +456,13 @@ struct _zend_op_array {
 	zend_string *doc_comment;
 
 	int last_literal;
+	uint32_t num_dynamic_func_defs;
 	zval *literals;
 
+	/* Functions that are declared dynamically are stored here and
+	 * referenced by index from opcodes. */
+	zend_op_array **dynamic_func_defs;
+
```

Comment:
Do we need "dynamic_func_defs" for each op array? or only for "main"?
May be we may store then in "literals"?

Reply:
> Do we need "dynamic_func_defs" for each op array? or only for "main"?

We need it for each op_array, as all op_arrays can contain closures or dynamically defined functions.

> May be we may store then in "literals"?

Storing the functions in `literals` was my first attempt (#5593), but I found the implementation more complicated (though this may be due to unnecessary refcounted wrappers I added).